src/intersections/triangles.jl

```diff
@@ -122,9 +122,70 @@ function intersecttype(f::Function, s::Segment{3,T}, t::Triangle{3,T}) where {T}
       # rejection 1, coplanar segment
       return NoIntersection() |> f
     end
-  end      
+  end
 
   λ = clamp(wᵥ / (wᵥ - sᵥ), zero(T), one(T))
 
   return IntersectingSegmentTriangle(s(λ)) |> f
-end
\ No newline at end of file
+end
+
+"""
+    intersecttype(s, t)
+
+Calculate the intersection of a ray and triangle in 3D
+and apply function `f` to it.
+
+## References
+
+* Möller, T. and Trumbore, B., 1997. [Fast, minimum storage ray-triangle
+  intersection. Journal of graphics tools]
+  (https://www.tandfonline.com/doi/abs/10.1080/10867651.1997.10487468)
+"""
+function intersecttype(f::Function, r::Ray{3,T}, t::Triangle{3,T}) where {T}
+  vₜ = vertices(t)
+  origin = r.p
+  D = r.v
+
+  E₁ = vₜ[3] - vₜ[1]
+  E₂ = vₜ[2] - vₜ[1]
+  P = D × E₂
+  det = E₁ ⋅ P
+
+  # keep det > 0, modify T accordingly
+  if det > atol(T)
+    TT = origin - vₜ[1]
```

Comment:
are you following the notation in the paper? we only use these syngle letters in algorithms when they follow the paper.

Reply:
It should be called `T`. But `T` used for type, so.. I will try to change to another letter.

Reply:
The paper uses `T` for this variable then? Interesting :) We just need to clear things up for future readers. We either choose single letters when the notation follows the notation of the paper or more explicit names like `dir` and `orig` when the paper doesn't provide notation.